src/job_worker/Worker.jl

```diff
@@ -0,0 +1,424 @@
+using Dates
+using Random
+using Logging
+using JSON3
+
+include("Config.jl")
+include("ECS.jl")
+include("HttpClient.jl")
+include("Jobs.jl")
+include("Storage.jl")
+
+"""
+Represents a job that needs to be processed
+"""
+struct Job
+    id::Int64
+    type::String
+    # We don't really know what type of data this will be 
+    input_payload::Any
+end
+
+"""
+Represents an assignment for a job
+"""
+struct JobAssignment
+    # ID of the assignment
+    id::Int64
+    # ID of the tasked job
+    job_id::Int64
+    # Path to where the data can be stored (in s3)
+    storage_uri::String
+end
```

Comment:
FYI, use strings to document fields if you really want, but otherwise leave as is.

https://docs.julialang.org/en/v1/manual/documentation/#Types

(yes, I too think it's a bit ugly)

Reply:
yeah yuck - have updated